### DISCUSSION TITLE
What static code analyzers are available for WGSL kernels?

### DISCUSSION THREAD

**Discussion by davids91** [gpuweb/gpuweb#4832] [Q&A]
I've found a [vscode plugin](https://github.com/wgsl-analyzer/wgsl-analyzer), but It's more like an indexer, is there anything available to check for simple problem -analysis and/or something to detect problems with?

**Comment by ferminLR** [comment]
Not sure you will find this useful, but can take a look at https://github.com/brendan-duncan/wgsl_reflect

**Reply by greggman** [reply]
AFAIK wgsl_reflect does not check for simple problems. In fact it has almost no error checking what-so-ever.

**Reply by ferminLR** [reply]
Yes it's a very limited tool. That's why I wasn't sure OP would find it useful; but decided to share it anyway.

**Comment by greggman** [comment]
I don't think anything exists but it would be helpful to know more concretely what you're hoping for. You can compile a WGSL shader in WebGPU and WebGPU will give you info. 

There's [tintd](https://marketplace.visualstudio.com/items?itemName=gfx-tintd.tintd) which does the same compilation that WebGPU does and will show errors. But, it needs some TLC as it doesn't do completions.

**Reply by davids91** [reply]
The rust equivalent of what I am looking for would be Clippy
